damnit/api.py

```diff
@@ -0,0 +1,250 @@
+from enum import Enum
+from pathlib import Path
+from contextlib import contextmanager
+
+import h5py
+import pandas as pd
+import xarray as xr
+
+from extra_data.read_machinery import find_proposal
+
+from .backend.db import DamnitDB, BlobTypes
+
+
+# This is a copy of damnit.ctxsupport.ctxrunner.DataType, purely so that we can
+# avoid the dependencies of the runner in the API (namely requests and pyyaml).
+class DataType(Enum):
+    DataArray = "dataarray"
+    Dataset = "dataset"
+    Image = "image"
+    Timestamp = "timestamp"
+
+
+class VariableData:
+    def __init__(self, name: str, proposal: int, run: int,
+                 h5_path: Path, data_format_version: int,
+                 db: DamnitDB, db_only: bool):
+        self._name = name
+        self._proposal = proposal
+        self._run = run
+        self._h5_path = h5_path
+        self._data_format_version = data_format_version
+        self._db = db
+        self._db_only = db_only
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def proposal(self):
+        return self._proposal
+
+    @property
+    def run(self):
+        return self._run
```

Comment:
I typically don't bother with making properties just to make attributes read-only, so e.g. `KeyData.key` and `.source` are regular attributes. I don't think we've had problems with people trying to change things like that.

Of course, there's no particular downside to making them read-only, it just makes the code a bit longer, so feel free to keep this if you prefer.

Reply:
True, but I think I'll keep them as properties so that I can document them more easily.

Reply:
OK :+1: